File: backend/src/main/java/com/equisoft/dca/backend/dataobject/service/DataObjectService.java
package com.equisoft.dca.backend.dataobject.service;

import java.util.List;
import java.util.Optional;

import javax.validation.Valid;

import com.equisoft.dca.backend.dataobject.model.DataObject;
import com.equisoft.dca.backend.dataobject.model.DataObjectFile;
import com.equisoft.dca.backend.dataobject.model.DataObjectType;
import com.equisoft.dca.backend.location.model.Location;
import com.equisoft.dca.backend.system.model.System;

public interface DataObjectService {

	/*
	DataObject update(@Valid DataObject dataObject);
*/
	DataObject save(@Valid DataObject dataObject);

	List<DataObject> save(String userIdentifier, List<DataObject> dataObjects);

	List<DataObject> create(System system, DataObjectType dataObjectType, List<DataObjectFile> dataObjectFiles);

	void deleteById(Integer id);

	Optional<DataObject> findById(Integer id);

	List<DataObject> findAll();

	List<DataObjectFile> validateDataObjects(DataObjectType dataObjectType, Location location,
			List<DataObjectFile> dataObjectFiles);
}

File: react/src/lib/services/DataObjectService.ts
import DataObject from '../domain/entities/DataObject';
import DataObjectContainer from '../domain/entities/DataObjectContainer';
import DataObjectFile from '../domain/entities/DataObjectFile';
import DataObjectRepository from '../domain/repositories/DataObjectRepository';

export default class DataObjectService {
    constructor(private dataObjectRepository: DataObjectRepository) {
    }

    async create(container: DataObjectContainer): Promise<DataObject[]> {
        return this.dataObjectRepository.create(container);
    }

    async import(dataObjects: DataObject[]): Promise<DataObject[]> {
        return this.dataObjectRepository.import(dataObjects);
    }

    async validate(dataObjectContainer: DataObjectContainer): Promise<DataObjectFile[]> {
        return this.dataObjectRepository.validate(dataObjectContainer);
    }

    async update(dataObject: DataObject): Promise<void> {
        return this.dataObjectRepository.update(dataObject);
    }

    async delete(id: string): Promise<void> {
        return this.dataObjectRepository.delete(id);
    }

    async findAll(): Promise<DataObject[]> {
        return this.dataObjectRepository.findAll();
    }

    async findById(id: string): Promise<DataObject> {
        return this.dataObjectRepository.findById(id);
    }
}
